specification/communication/Communication.Messages/models.tsp

```diff
@@ -0,0 +1,322 @@
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.Core.Traits;
+using Azure.ClientGenerator.Core;
+
+@versioned(AzureCommunicationMessagesService.Versions)
+namespace AzureCommunicationMessagesService;
+
+@access(Access.public)
+@doc("The type of message. Supports text, image, template.")
+enum CommunicationMessageType {
+  @doc("Text message type.")
+  text,
+
+  @doc("Image message type.")
+  image,
+
+  @doc("Template message type.")
+  template,
+}
+
+@access(Access.public)
+@doc("The WhatsApp button sub type.")
+enum WhatsAppMessageButtonSubType {
+  @doc("The WhatsApp button sub type is quick reply.")
+  quickReply,
+
+  @doc("The WhatsApp button sub type is url.")
+  url,
+}
+
+@access(Access.public)
+@doc("The aggregated template status.")
+enum MessageTemplateStatus {
+  @doc("Message template is approved.")
+  approved,
+
+  @doc("Message template is rejected.")
+  rejected,
+
+  @doc("Message template is pending.")
+  pending,
+
+  @doc("Message template is paused.")
+  paused,
+}
+
+@discriminator("kind")
+@doc("Details of the message to send.")
+model NotificationContent {
+  @doc("The Channel Registration ID for the Business Identifier.")
+  channelRegistrationId: uuid;
+
+  @doc("The native external platform user identifiers of the recipient.")
+  to: string[];
+}
+
+@doc("A request to send a text notification.")
+model TextNotificationContent extends NotificationContent {
+  @doc("Message notification type is text.")
+  kind: "text";
+
+  @doc("Message content.")
+  content: string;
+}
+
+@doc("A request to send a media notification.")
+model MediaNotificationContent extends NotificationContent {
+  @doc("Message notification type is image.")
+  kind: "image";
+
+  @doc("Optional text content.")
+  content?: string;
+
+  @doc("""
+A media url for the file. Required if the type is one of the supported media
+types, e.g. image
+""")
+  mediaUri: url;
+}
+
+@doc("A request to send a template notification.")
+model TemplateNotificationContent extends NotificationContent {
+  @doc("Message notification type is template.")
+  kind: "template";
+
+  @doc("The template object used to create templates.")
+  template: MessageTemplate;
+}
+
+@doc("The template object used to create templates.")
+model MessageTemplate {
+  @doc("Name of the template.")
+  name: string;
+
+  @doc("The codes for the supported languages for templates.")
+  language: string;
+
+  @doc("The template values.")
+  values?: MessageTemplateValue[];
+
+  @doc("The binding object to link values to the template specific locations")
+  bindings?: MessageTemplateBindings;
+}
+
+@discriminator("kind")
+@doc("The class describes a parameter of a template.")
+model MessageTemplateValue {
+  @doc("Name of the Template value")
+  name: string;
+}
+
+@doc("The message template's text value information.")
+model MessageTemplateText extends MessageTemplateValue {
+  @doc("Message parameter type is text.")
+  kind: "text";
```

Comment:
Should we extract the literal out to an `enum MessageTemplateValueType`? Is there a difference in the generated SDK when we use a string literal vs an enum or do the APIs look the same?

Reply:
was extracted